jpa/src/main/java/org/jdbi/v3/jpa/BindJpaFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.jpa;
+
+import org.jdbi.v3.sqlobject.Binder;
+import org.jdbi.v3.sqlobject.BinderFactory;
+
+public class BindJpaFactory implements BinderFactory<BindJpa, Object> {
+    @Override
+    public Binder<BindJpa, Object> build(BindJpa annotation) {
+        return (q, parameter, bind, arg) -> {
+            final String prefix;
+            if (BindJpa.DEFAULT.equals(bind.value())) {
+                prefix = "";
+            } else {
+                prefix = bind.value() + ".";
+            }
+
+            try {
+                JpaClass<?> jpaClass = JpaClass.get(arg.getClass());
+                for (JpaMember member : jpaClass.members()) {
+                    q.dynamicBind(member.getType(), prefix + member.getColumnName(), member.read(arg));
+                }
+            } catch (Exception e) {
+                throw new IllegalStateException(
```

Comment:
Should this wrap subclasses of `RuntimeException`? Will end up wrapping e.g. `DbiException` and friends.